crates/runtime/src/instance/allocator.rs

```diff
@@ -531,14 +531,25 @@ fn initialize_tables(instance: &mut Instance, module: &Module) -> Result<()> {
             TableInitialValue::FuncRef(idx) => {
                 let funcref = instance.get_func_ref(*idx).unwrap();
                 let table = unsafe { &mut *instance.get_defined_table(table) };
-                table.init_func(funcref)?;
+                let init = (0..table.size()).map(|_| funcref);
+                table.init_func(0, init)?;
             }
 
             TableInitialValue::GlobalGet(idx) => unsafe {
                 let global = instance.defined_or_imported_global_ptr(*idx);
-                let funcref = (*global).as_func_ref();
                 let table = &mut *instance.get_defined_table(table);
-                table.init_func(funcref)?;
```

Comment:
Relative to https://github.com/bytecodealliance/wasmtime/pull/8016 this was the other issue I found, this forgot to check whether the table was an externref table or funcref table and triggered a panic by assuming a global was a funcref.